plugins/accelerated-peft/src/fms_acceleration_peft/autogptq_utils.py

```diff
@@ -35,11 +35,28 @@
 PEFT_ALL_LINEAR = "all-linear"
 
 
-def requires_installation_on_all_linears(peft_config):
+def requires_installation_on_all_linears(peft_config, model_type: str = None):
     tm = peft_config.target_modules
+
+    if tm is None:
+        try:
+            # Third Party
+            # pylint: disable=import-outside-toplevel
+            from peft.utils.constants import (
+                TRANSFORMERS_MODELS_TO_LORA_TARGET_MODULES_MAPPING,
+            )
+
+            tm = TRANSFORMERS_MODELS_TO_LORA_TARGET_MODULES_MAPPING[model_type]
+        except (ImportError, IndexError) as e:
```

Comment:
If the model_type isn't able to be found then it will pass it in as None and then when trying to do `TRANSFORMERS_MODELS_TO_LORA_TARGET_MODULES_MAPPING[None]` this leads to a KeyError. I think this should also be captured in this error. The same would happen if a model_type is given that is not in the dictionary. I don't know if we would hit IndexError

Reply:
Ah thanks for his catch. i intended to use `KeyError` but somehow mistook it as an `IndexError`. The latter is for arrays but here we are dealing with a dict